### PR TITLE
feature(frontend): environment ux improvements

### PR DESCRIPTION
**Pull request by xoscar**
This PR implements the Environments UX/UI changes described here [Environments page improvement#1986](https://github.com/kubeshop/tracetest/issues/1986)

## Changes

- Moves the environment modal to its own component folder
- Extends the environment provider to include the CRUD
- Decouples the environment modal from the API calls

## Fixes

- #2019 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/c6e25f981e1b4cabbce97ac5738bcc8e

### CODE REVIEW

File: web/src/providers/Environment/hooks/useEnvironmentCrud.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        onConfirm: () => deleteEnvironment({environmentId: id}),
+      });
+    },
+    [deleteEnvironment, onOpen]
+  );
+
+  const edit = useCallback(
+    async (environmentId: string, environment: Environment) => {
+      await updateEnvironment({environmentId, environment});
+      message.success('Environment updated successfully');
```
Comment (jorgeepc) [inline web/src/providers/Environment/hooks/useEnvironmentCrud.ts:30]: We should try to use an standard component for those confirmation messages. I think we have a different one for DataStores. Not a blocker we can handle that in a different task.

**Review by jorgeepc — APPROVED** [review]
Clean 🧼  Great job dude!

File: web/src/providers/Environment/hooks/useEnvironmentCrud.ts (hunk 1 above)
Comment (olha23) [reply web/src/providers/Environment/hooks/useEnvironmentCrud.ts:30]: agree with @jorgeepc comment

Comment (xoscar) [reply web/src/providers/Environment/hooks/useEnvironmentCrud.ts:30]: I believe we added the messages to provide feedback for some of the FE actions. 

I created this issue a long time ago while thinking about adding a standardized notification system https://github.com/kubeshop/tracetest/issues/1478

Let's all think about that and how we can introduce it in more general way 😄

**Pull request merged**